tgui/packages/tgui/interfaces/PayStand.tsx

```diff
@@ -0,0 +1,238 @@
+            <Stack.Item>
+              <AccountDisplay onClick={onClick} />
+            </Stack.Item>
+            <Stack.Item grow>
+              {!setupMode ? (
+                <TerminalDisplay onClick={onClick} />
+              ) : (
+                <SetupDisplay onClick={onClick} />
+              )}
+            </Stack.Item>
+          </Stack>
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+/**
+ * Displays the current user's bank information (if any)
+ */
+const AccountDisplay = (_, context) => {
+  const { data } = useBackend<PayStandData>(context);
+  const { user } = data;
+  if (!user) {
+    return <NoticeBox>Error! No account detected.</NoticeBox>;
+  }
+
+  return (
+    <Section>
+      <Table fill>
+        <Table.Row>
+          <Table.Cell>
+            <Box color="label">
+              <Icon name="money-check" color="label" mr={1} />
+              {user?.name}
+            </Box>
+          </Table.Cell>
+          <Table.Cell collapsing>
+            <Box color="good">
+              {user?.balance} cr <Icon color="gold" name="coins" />
+            </Box>
+          </Table.Cell>
+        </Table.Row>
+      </Table>
+    </Section>
+  );
+};
+
+/**
+ * Displays the payment processor. This is the main display.
+ * Shows icon, name, payment button.
+ */
+const TerminalDisplay = (props, context) => {
+  const { act, data } = useBackend<PayStandData>(context);
+  const { description, force_fee, name, owner, user, shop_logo } = data;
+  const { onClick } = props;
+  const is_owner = owner === user?.name;
+  const cannot_pay
+    = is_owner || !user || user?.balance < 1 || user?.balance < force_fee;
+  const decodedName = name.replace(/&#(\d+);/g, (_, dec) => {
+    return String.fromCharCode(dec);
+  });
+
+  return (
+    <Section
+      buttons={
+        is_owner && (
+          <Button icon="edit" onClick={onClick}>
+            Setup
+          </Button>)
+      }
+      fill
+      title="Terminal">
+      <Stack fill vertical>
+        <Stack.Item align="center">
+          <Icon color="good" name={shop_logo} size="5" />
+        </Stack.Item>
+        <Stack.Item grow textAlign="center">
+          <Tooltip content={description} position="bottom">
+            <Box color="label" fontSize="17px" overflow="hidden">
+              {decodedName}
+            </Box>
+          </Tooltip>
+        </Stack.Item>
+        <Stack.Item>
+          <Button
+            disabled={cannot_pay}
+            fluid
+            height="2rem"
+            onClick={() => act('pay')}
+            pt={0.2}
+            textAlign="center">
+            <Icon name="coins" />
+            Pay {!!force_fee && force_fee + ' cr'}
+          </Button>
+        </Stack.Item>
+        <Stack.Item>
+          {/* @ts-ignore */}
+          <marquee scrollamount="2">
+            <Box color="darkgray" fontSize="8px">
+              {COPYRIGHT_SCROLLER}
+            </Box>
+            {/* @ts-ignore */}
+          </marquee>
```

Comment:
Wait, what, is this an invalid element in TypeScript?

Reply:
it throws an error! which is strange because i got this idea from your error screen

Reply:
You could play around with JSX.IntrinsicElements in global.d.ts to make IE exclusive elements work in TypeScript

```
declare global {
  namespace JSX {
    interface IntrinsicElements {
      'xx-element1': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>; // Normal web component
      'xx-element2': React.DetailedHTMLProps<React.HTMLAttributes<HTMLInputElement>, HTMLInputElement>; // Web component extended from input
    }
  }
}
```

https://stackoverflow.com/questions/61015445/using-web-components-within-preact-and-typescript